app/src/backlight.c

```diff
@@ -42,56 +39,41 @@ static struct backlight_state state = {.brightness = CONFIG_ZMK_BACKLIGHT_BRT_ST
 
 static int zmk_backlight_update() {
     uint8_t brt = state.on ? state.brightness : 0;
-    for (int i = 0; i < BACKLIGHT_NUM_LEDS; i++) {
-        int rc = led_set_brightness(backlight_dev, i, brt);
-        if (rc != 0) {
-            return rc;
-        }
-    }
-    return 0;
+    return led_set_brightness(backlight_dev, 0, brt);
 }
 
 #if IS_ENABLED(CONFIG_SETTINGS)
 static int backlight_settings_set(const char *name, size_t len, settings_read_cb read_cb,
                                   void *cb_arg) {
     const char *next;
+    int rc;
 
     if (settings_name_steq(name, "state", &next) && !next) {
         if (len != sizeof(state)) {
             return -EINVAL;
         }
 
-        int rc = read_cb(cb_arg, &state, sizeof(state));
-        if (rc < 0) {
-            return rc;
+        rc = read_cb(cb_arg, &state, sizeof(state));
+        if (rc >= 0) {
+            return 0;
         }
 
-        return zmk_backlight_update();
+        return rc;
```

Comment:
`zmk_backlight_update` must be called after the settings are loaded. Why did you change it?

Reply:
settings_set doesn't load the settings thus it doesn't need to update the lights, it saves the current settings. The settings are loaded in backlight_init

Reply:
According to the documentation ([https://docs.zephyrproject.org/latest/reference/settings/index.html#handlers](https://docs.zephyrproject.org/latest/reference/settings/index.html#handlers)), `h_set` is called "_when the value is loaded from persisted storage_", the settings are saved using `settings_save_one`.

Reply:
I was under the impression that the loading update would be caught in the backlight init's call of zmk_backlight_update. Looking again I see it submits a delayed work request so it might get missed by that call. putting it in here too is correct, sorry, will fix

Reply:
In my testing the settinigs load call was executed quickly enough that the update in the init worked every time however given more complicated features in the future definitely worth keeping. I was trying to minimise spurious function calls